pkg/importer/http-datasource.go

```diff
@@ -395,3 +420,67 @@ func parseHTTPHeader(resp *http.Response) uint64 {
 
 	return total
 }
+
+// Check for any extra headers to pass along. Return secret headers separately so callers can suppress logging them.
+func getExtraHeaders() ([]string, []string, error) {
+	extraHeaders := getExtraHeadersFromEnvironment()
+	secretExtraHeaders, err := getExtraHeadersFromSecrets()
+	return extraHeaders, secretExtraHeaders, err
+}
+
+// Check for extra headers from environment variables.
+func getExtraHeadersFromEnvironment() []string {
+	var extraHeaders []string
+
+	for _, value := range os.Environ() {
+		if strings.HasPrefix(value, common.ImporterExtraHeader) {
+			env := strings.SplitN(value, "=", 2)
+			if len(env) > 1 {
+				extraHeaders = append(extraHeaders, env[1])
+			}
+		}
+	}
+
+	return extraHeaders
+}
+
+// Check for extra headers from mounted secrets.
+func getExtraHeadersFromSecrets() ([]string, error) {
+	var secretExtraHeaders []string
+	var err error
+
+	secretDir := common.ImporterSecretExtraHeadersDir
+	directories, err := ioutil.ReadDir(secretDir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return secretExtraHeaders, nil
+		}
+		return nil, errors.Wrapf(err, "Error listing directories under %s", secretDir)
+	}
+
+	for _, directory := range directories {
```

Comment:
would a filepath.Walk make this easier to read?

Reply:
Sure, I tried it out and I think it looks a little cleaner. It is slightly different because it goes through the whole secrets directory recursively instead of only one level deep, but I don't think this should cause any practical trouble.